kernel/samples/bpf/xdp_ddos01_blacklist_cmdline.c

```diff
@@ -197,21 +210,51 @@ static void blacklist_print_ip(__u32 ip, __u64 count)
 	printf("\"%s\" : %llu\n", ip_txt, count);
 }
 
+static void blacklist_print_port(int key, u32 val, __u64 count)
+{
+	int i;
+	printf(" %d: ", key);
+	for (i = 0; i < XDP_PROTO_FILTER_MAX; i++)
+		if (val & (1 << i))
+			printf("%s ",xdp_proto_filter_names[i]);
+	printf(": %llu\n", count);
+}
+
 static void blacklist_list_all(int fd)
 {
 	__u32 key = 0, next_key;
 	__u64 value;
 
 	printf("{\n");
 	while (bpf_map_get_next_key(fd, &key, &next_key) == 0) {
-		printf("%s", key ? "," : " ");
```

Comment:
Sorry that was accidental.

I'll add back as I also have some thoughts on adding a --port and --address args to --list so that the output is JSON for both.